engine/include/cubos/engine/voxels/model.hpp

```diff
@@ -0,0 +1,71 @@
+/// @file
+/// @brief Class @ref cubos::engine::VoxelModel
+/// @ingroup voxels-plugin
+
+#pragma once
+
+#include <vector>
+
+#include <cubos/core/reflection/reflect.hpp>
+
+#include <cubos/engine/assets/bridges/file.hpp>
+#include <cubos/engine/voxels/grid.hpp>
+#include <cubos/engine/voxels/palette.hpp>
+
+namespace cubos::engine
+{
+    /// @brief Holds the data loaded from voxel model files, such as Qubicle models.
+    ///
+    /// @ingroup voxels-plugin
+    class CUBOS_ENGINE_API VoxelModel final
+    {
+    public:
+        CUBOS_REFLECT;
+
+        ~VoxelModel() = default;
+
+        /// @brief Constructs an empty VoxelModel
+        VoxelModel() = default;
+
+        std::size_t gridCount() const;
+
+        const VoxelPalette& palette() const;
+
+        VoxelPalette& palette();
+
+        const VoxelGrid& grid(std::size_t index) const;
+
+        VoxelGrid& grid(std::size_t index);
+
+        glm::ivec3 gridPosition(std::size_t index) const;
+
+        void gridPosition(std::size_t index, glm::ivec3 position);
```

Comment:
Make sure to document these methods so that they appear in the documentation website!